substrate/frame/revive/src/primitives.rs

```diff
@@ -391,6 +392,9 @@ pub struct ExecConfig<T: Config> {
 	/// This is primarily used for testing purposes and should be `None` in production
 	/// environments.
 	pub mock_handler: Option<Box<dyn MockHandler<T>>>,
+	/// external transient storage useful for testing. should be `None`  in production
+	/// environments.
+	pub transient_storage: Option<Rc<RefCell<TransientStorage<T>>>>,
```

Comment:
Move into the `Stack` using `take()` when the Stack is created.

```suggestion
	pub initial_transient_storage: Option<TransientStorage<T>>>,
```

Reply:
we still need to have it back in foundry, so moving into into the stack will not allow us to get the instance back

Reply:
Ahh, ok